### PR TITLE
CC-6659: Auto build toolkit on changes

### PR DESCRIPTION
**Pull request by ellacai**
### :hammer_and_wrench: Description
Now you can do a `yarn watch:toolkit` which will watch for changes in `toolkit/src` and automatically rebuild so the `documentation` app will have the latest changes.
Before, you had to run `yarn prepare` after each toolkit change.

### :camera_flash: Screenshots

https://github.com/hashicorp/consul-ui-toolkit/assets/23414052/8f64e7d5-e6b6-45f5-845a-e849dc48ab8f

### :link: External Links
[jira](https://hashicorp.atlassian.net/browse/CC-6659)

### :+1: Definition of Done

- [ ] New functionality works?
- [ ] Tests added?
- [ ] Docs updated?

### :speech_balloon: Using the [Netlify feedback ladder](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify/)

- :mount_fuji: **[mountain]**: Blocking and requires immediate action.
- :moyai: **[boulder]**: Blocking.
- :white_circle: **[pebble]**: Non-blocking, but requires future action.
- :hourglass_flowing_sand: **[sand]**: Non-blocking, but requires future consideration.
- :sparkles: **[dust]**: Non-blocking. "Take it or leave it"

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,8 +39,8 @@
 
 ## Running the test application
 
-- Run `yarn prepare` after any changes to the toolkit to build it to be consumed by the documentation app.
-- Run `yarn start:documentation` to start the documentations apps server
+- Run `yarn watch:toolkit` - This will automatically detect changes in the toolkit and build it to be consumed by the documentation app.
+- In a separate terminal, run `yarn start:documentation` to start the documentations apps server
```
Comment (WenInCode) [inline CONTRIBUTING.md:43]: might be a good idea to mention that this is using `fb-watchman` and maybe link to their docs.

**Review by WenInCode — APPROVED** [review]
Seems to work great. Only thing that isn't working for me is if I create a new component it doesn't seem to work in the docs unless I kill it and re prepare. I am not sure what is the cause. Maybe try looking into it a bit more, but this is still a nice improvement.

**Comment by ellacai** [comment]
> Seems to work great. Only thing that isn't working for me is if I create a new component it doesn't seem to work in the docs unless I kill it and re prepare. I am not sure what is the cause. Maybe try looking into it a bit more, but this is still a nice improvement.

hmm if you create a new component, wouldn't the `documentation` app throw a `Attempted to resolve `cut/new-component`, which was expected to be a component, but nothing was found.` error that requires a kill and re-build anyways? 

maybe i'm misunderstanding but I thought that was expected ❓

**Comment by WenInCode** [comment]
@ellacai Try this flow

1. Start the documentation app
2. Add a simple component to toolkit like `Cut::Fake` that just renders a div with some text
3. Run `yarn prepare`
4. Try to use that component in documentation without restarting the build.

I'm pretty sure it will work 🤔

**Comment by ellacai** [comment]
> @ellacai Try this flow
> 
> 1. Start the documentation app
> 2. Add a simple component to toolkit like `Cut::Fake` that just renders a div with some text
> 3. Run `yarn prepare`
> 4. Try to use that component in documentation without restarting the build.
> 
> I'm pretty sure it will work 🤔

I gave it a try and I think I get the same error? 
We can revisit this in a future improvement! Will merge as is for now.

**Pull request merged**